gh_actions_exporter/cost.py

```diff
@@ -0,0 +1,123 @@
+        duration: float = (job_request["completed_at"].timestamp()
+                    - job_request["started_at"].timestamp())
+
+        return duration / 60 * cost_per_min
+
+    def get_previous_check_run(self, repository: Repository, webhook: WebHook) -> str:
+        url: str = f"/repos/{webhook.repository.full_name}/commits/{webhook.workflow_run.head_sha}/check-runs"
+        check_runs: dict = repository._requester.requestJsonAndCheck("GET", url, {"Accept": "application/vnd.github.v3+json",})[1]
+
+        for check_run in check_runs["check_runs"]:
+            if check_run["name"] == "self.settings.title" and check_run["app"]["id"] == self.settings.github_app_id:
+                return check_run["output"]["summary"]
+        return ""
+
+    def upload_check_run(self, summary: str, webhook: WebHook, access_token: str) -> None:
+        g: Github = Github(login_or_token=access_token)
+        repo: Repository = g.get_repo(webhook.repository.full_name)
+
+        repo.create_check_run(
+            name="Cost",
+            head_sha=webhook.workflow_run.head_sha,
+            status="completed",
+            conclusion="success",
+            output={
+                "title": self.settings.title,
+                "summary": summary
+            }
+        )
+
+    def display_cost(self, webhook: WebHook):
+        if webhook.workflow_run.conclusion:
+            token_generator: TokenGenerator = TokenGenerator(self.settings)
+            access_token: str = token_generator.generate_token()
+            
+            g: Github = Github(access_token)
+            repository: Repository = g.get_repo({webhook.repository.full_name})
+            url: str = f"/repos/{webhook.repository.full_name}/actions/runs/{webhook.workflow_run.id}/jobs"
+            workflow_jobs: dict = repository._requester.requestJsonAndCheck("GET", url, {"Accept": "application/vnd.github.v3+json",})[1]
+
+            total_cost: float = 0.0
+            jobs_cost: List[JobCost] = []
+            for job in workflow_jobs["jobs"]:
+                cost: float = self._get_job_cost(job, webhook)
+                jobs_cost.append(JobCost(name=job["name"], cost=cost))
+                total_cost += cost
+
+            summary: str = self.get_previous_check_run(repository, webhook)
+
+            template = jinja2.Template(
+                """{% if summary == "" %}
+                {{ settings.summary }}
+                {% endif %}
+                ## [{{ workflow_run.name }}]({{ workflow_run.html_url }})
+                {% for job_cost in jobs_cost %}* {{ job_cost.name }}: ${{ job_cost.cost }}
+                {% endfor %}Total: ${{ total_cost }}"""
+            )
+
+            data: dict = {
+                "summary": summary,
+                "settings": {"summary": self.settings.summary},
+                "workflow_run": {"name": webhook.workflow_run.name, "html_url": webhook.workflow_run.html_url},
```

Comment:
why not?

I understand you only want to mention what you need, but I see no harm in keeping the whole workflow run object available if needed.
```suggestion
                "workflow_run": webhook.workflow_run,
```